### ISSUE TITLE
`dvc pull`: Config filed `jobs` in `.dvc/config` not taken into account as expected

### ISSUE THREAD

**Issue by stefan-hartmann-lgs** [iterative/dvc#7730]
# Bug Report

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor

DVC version: 2.10.2 (exe)
---------------------------------
Platform: Python 3.8.10 on Windows-10-10.0.19042-SP0
Supports:
        azure (adlfs = 2022.4.0, knack = 0.9.0, azure-identity = 1.10.0),
        gdrive (pydrive2 = 1.10.1),
        gs (gcsfs = 2022.3.0),
        hdfs (fsspec = 2022.3.0, pyarrow = 8.0.0),
        webhdfs (fsspec = 2022.3.0),
        http (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
        https (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
        s3 (s3fs = 2022.3.0, boto3 = 1.21.21),
        ssh (sshfs = 2022.3.1),
        oss (ossfs = 2021.8.0),
        webdav (webdav4 = 0.9.7),
        webdavs (webdav4 = 0.9.7)
Cache types: <https://error.dvc.org/no-dvc-cache>
Caches: local
Remotes: https
Workspace directory: NTFS on C:\
Repo: dvc, git
```

**Comment by daavoo** [comment]
This is not currently supported (see available config fields for `remote` section in https://dvc.org/doc/command-reference/config#remote).

Is there any particular part of the documentation that led you to:

```
It was expected that the jobs=4 config in .dvc/config file does the same as running dvc pull --jobs 4 which is not the case.
```

**Comment by dberenbaum** [comment]
@daavoo That links to https://dvc.org/doc/command-reference/remote/modify#available-parameters-for-all-remotes, which does show `jobs` as an option. Am I misunderstanding?

**Comment by daavoo** [comment]
🙏 ignore muy previous comment

**Comment by stefan-hartmann-lgs** [comment]
> @daavoo That links to https://dvc.org/doc/command-reference/remote/modify#available-parameters-for-all-remotes, which does show `jobs` as an option. Am I misunderstanding?

Yes exactly. The `jobs` in `.dvc/config` does something but not the same what `--jobs` does an that was expected as stated in the documentation.

**Comment by daavoo** [comment]
Hi @stefan-hartmann-lgs ! How are you checking the `See that more than 4 download jobs are running in parallel`??

I have tried and the `jobs` config section is honored when instantiating the filesystem and passed down to the transferring task :

https://github.com/iterative/dvc/blob/f23d31af644ab4ad4492b9cfa1000d58420c238d/dvc/fs/__init__.py#L94-L110

Could you try sharing the output profile from:

```bash
pip install viztracer
dvc pull --viztracer-depth 8
```

**Comment by stefan-hartmann-lgs** [comment]
Hi @daavoo 

Sorry for laaaaate reply - I can see that more than 4 jobs are running in my console:

![image](https://user-images.githubusercontent.com/65395387/175943686-15b2ab54-1450-4007-b833-ea39f766b0a3.png)

**Comment by karajan1001** [comment]
It's better to use `dvc config -l` instead of type `type .dvc/config`. for there might be some hidden things in `.dvc/config.local`

**Comment by stefan-hartmann-lgs** [comment]
@daavoo any infos on this one?

**Comment by skshetry** [comment]
I tested this and can confirm that the `jobs` value from the config is being respected. Closing…
